### PR TITLE
feat(adirs): Improve ADIRS SSM and behaviours

### PR DESCRIPTION
**Pull request by lukecologne**
## Summary of Changes

This PR:
* Outputs SSM=FW for all ADR labels when the ADR is off or not aligned
* Improves the ADR behaviour for low speeds. The following values are now coded as NCD with value=0 when below a threshold:
   * CAS, when below 30kts
   * TAS, when below 60kts
   * Mach, when below 0.1
* All 3 ADRs now output temperature values, instead of only 1 and 3.
* Added new data outputs to the ADR:
   * Angle of Attack (coded NCD below 60kts CAS, but the actual value is still output)
* Added new data outputs to the IR:
   * Drift Angle (DA), which is 0 below 50kts GS
   * Kinematic Flight Path Angle (gamma_k), which is 0 below 50kts GS
   * The Body axis rotation rates: p, q, r
      * Derived from these the Pitch/Roll/Heading rates: Theta_dot, Phi_dot, Psi_dot
   * The Body accelerations: a_x, a_y, a_z

The FPA and DA are now used for the FPV on the PFD. In case of FPA or DA set as FW/NCD, the FPV flag is now shown (when the attitude is still valid and TRK/FPA mode is active, for example when the IR is in ATT mode).
Also, if the CAS goes below 30kts in flight, the SPD flag is shown now, since CAS is then coded NCD.
Finally, if mach is not valid, the MACH flag is not shown on ground, only in flight. This means that if MACH goes below 0.1 in the air, the MACH flight will also be shown, since it is coded NCD in that case.

## Screenshots (if necessary)

https://streamable.com/hudxhn

## References

For low airspeeds and extreme attitudes, I found these videos, in which you can clearly see the SPD and MACH flag, and also the FPV snapping away below 50kts:
https://www.youtube.com/watch?v=Eap-3siSgLc
https://www.youtube.com/watch?v=rKEKwYTvuZU
https://www.youtube.com/watch?v=AYm1aLu49Pk

## Additional context

Discord username (if different from GitHub):

## Testing instructions

* Make sure that the bird and the FPV flag on the PFD in TRK/FPA mode beave as described above and are plausible
* Make sure that the CAS/TAS/mach values (and invalidity flags) on the ND/PFD behave as described above and are plausible
* Unfortunately, the other parameters added (AoA, the rotation rates and accelerations) cannot be easily checked without decoding the Arinc429 values and looking at the numeric value, so they probably cannot be tested properly.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: src/systems/systems/src/navigation/adirs.rs [hunk 1]
```diff
@@ -2261,6 +2380,48 @@ mod tests {
...
+            let heading = Angle::new::<degree>(160.);
+            let track = Angle::new::<degree>(180.);
+            let mut test_bed = all_adirus_aligned_test_bed_with()
+                .track_of(track)
+                .and()
+                .heading_of(heading)
+                .and()
+                .ground_speed_of(Velocity::new::<knot>(
+                    InertialReference::MINIMUM_GROUND_SPEED_FOR_TRACK_KNOTS,
+                ));
```
Comment (davidwalschots) [inline src/systems/systems/src/navigation/adirs.rs:2399]: **nitpick**: The goal of `and` here is simply to form English sentences. So probably you'd want to only keep the last one:
> All adirus aligned test bed with track of 123, heading of 124, and ground speed of 50.

**Review by davidwalschots — COMMENTED** [review]
Minor remark on a test. In general I don't do this, but this time I've created a PR here, which you could merge in if you deem it useful: https://github.com/lukecologne/a32nx/pull/1

**Review by davidwalschots — APPROVED** [review]
I've reviewed the Rust side of things. Looks good!

**Review by beheh — APPROVED** [review]

**Comment by ShadowFrost3** [comment]
Quality Assurance Trainee Report


Object of testing: #6636
Tier of Testing:1
Date: 08/03/2022

Testing Process: 
Full flight - passed

Make sure that the bird and the FPV flag on the PFD in TRK/FPA mode beave as described above and are plausible - no issues noted in normal operations 

Make sure that the CAS/TAS/mach values (and invalidity flags) on the ND/PFD behave as described above and are plausible - checked

Unfortunately, the other parameters added (AoA, the rotation rates and accelerations) cannot be easily checked without decoding the Arinc429 values and looking at the numeric value, so they probably cannot be tested properly. - not checked

Had a couple game crashes just need to verify they are not PR related - tomorrow 
Crashed on hotstart runway takeoff roll 

Negatives:
Testing Results: Passed with the condition that PR is not causing crashes. I find it unlikely..... but due diligence 


image

**Comment by mico975** [comment]
Quality Assurance Tester Report

Discord          : mico#3145
Object of testing: #6636
Tier of Testing  : 1
Date             : 10/03/2022

Testing Process: 
Make sure that the bird and the FPV flag on the PFD in TRK/FPA mode beave as described above and are plausible - OK
Make sure that the CAS/TAS/mach values (and invalidity flags) on the ND/PFD behave as described above and are plausible - OK

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
LGTM

Media:       
image

**Pull request merged**